src/components/Search/CompareButton.tsx

```diff
@@ -1,20 +1,48 @@
 import Button from '@mui/material/Button';
 
+import { useAppSelector } from '../../hooks/app';
+import { SearchStyles } from '../../styles';
+import CancelButton from './CancelButton';
+
 interface CompareButtonProps {
   label: string;
+  hasNonEditableState: boolean;
+  onCancel: () => void;
+  onSetDisplayForm: (display: boolean) => void;
+  formIsDisplayed: boolean;
 }
 
-export default function CompareButton({ label }: CompareButtonProps) {
+export default function CompareButton({
```

Comment:
What do you think of renaming to "CancelAndCompareButton"?

Reply:
Yes, I agree.